internal/report/create.go

```diff
@@ -0,0 +1,140 @@
+package report
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/kubewarden/audit-scanner/internal/constants"
+	policiesv1 "github.com/kubewarden/kubewarden-controller/pkg/apis/policies/v1"
+	"github.com/rs/zerolog/log"
+	admv1 "k8s.io/api/admission/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/wg-policy-prototypes/policy-report/pkg/api/wgpolicyk8s.io/v1beta1"
+)
+
+type PolicyReport v1beta1.PolicyReport
+type ClusterPolicyReport v1beta1.ClusterPolicyReport
+
+const PolicyReportSource string = "kubewarden"
+
+var clusterPolicyReport = ClusterPolicyReport{
+	ObjectMeta: metav1.ObjectMeta{Name: "polr-clusterwide"},
+	Summary: v1beta1.PolicyReportSummary{
+		Pass:  0, // count of policies with requirements met
+		Fail:  0, // count of policies with requirements not met
+		Warn:  0, // not used for now
+		Error: 0, // count of policies that couldn't be evaluated
+		Skip:  0, // count of policies that were not selected for evaluation
+	},
+	Results: []*v1beta1.PolicyReportResult{},
+}
+
+// slice of namespaced PolicyReports
+var policyReports = make(map[string]*PolicyReport)
+
+// New() creates a new PolicyReport for the corresponding namespace if it
+// doesn't exist, setting the number of skipped policies in the
+// PolicyReport{}.Summary.
+func New(namespace string, skippedNum int) error {
+	// policy report doesn't exist and needs to be created
+	if policyReports[namespace] == nil {
+		policyReports[namespace] = &PolicyReport{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:              "polr-ns-" + namespace,
+				Namespace:         namespace,
+				CreationTimestamp: metav1.Now(),
+			},
+			Scope: &v1.ObjectReference{}, // TODO to namespace
```

Comment:
this is missing; we have the `namespace` string. Ideally, report.New() should receive the Object already. Other backends (SQL) would be ok with a string I suppose.

Reply:
done in https://github.com/kubewarden/audit-scanner/pull/40/commits/bf47e0dea867579da05812a3adf1215f555c225f